src/v/net/connection_rate_counter.h

```diff
@@ -87,7 +87,7 @@ class connection_rate_counter {
 private:
     bool need_wait() { return avaiable_new_connections() == 0; }
 
-    ss::basic_semaphore<ss::semaphore_default_exception_factory, Clock> bucket;
+    ss::basic_semaphore<ss::named_semaphore_exception_factory, Clock> bucket;
```

Comment:
Does this deserve a typedef (or alias) in, for example, `ssx/named_semaphore.h`, something like, `ssx::named_timeout_semaphore`?

Reply:
Or maybe `ssx::timeout_semaphore` - since naming should be everywhere? But take some notes from whatever seastar calls this thing.

Reply:
Now that I think about it maybe all semaphores should be attached to a timeout, but that definitely deserves a separate commit / PR.

Reply:
> Now that I think about it maybe all semaphores should be attached to a timeout, but that definitely deserves a separate commit / PR.

giving a semaphore a timeout should probably be considered a special case

Reply:
Yeah. There's only one instance of this non-default clock instantiaiton so far. When there are >1 we could do this.